### PR TITLE
l3_interface rm_template based implementation and bug fixes

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

L3_interface rm_template based implementation and bug fixes

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

l3_interfaces

### CODE REVIEW

**Comment by KB-perByte** [comment]
Module | statements | missing | excluded | branches | partial | coverage
-- | -- | -- | -- | -- | -- | --
Total | 316 | 2 | 0 | 156 | 15 | 96%
plugins/module_utils/network/ios/argspec/l3_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/argspec/l3_interfaces/l3_interfaces.py | 4 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/config/l3_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/config/l3_interfaces/l3_interfaces.py | 128 | 0 | 0 | 106 | 11 | 95%
plugins/module_utils/network/ios/facts/l3_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/ios/facts/l3_interfaces/l3_interfaces.py | 43 | 1 | 0 | 18 | 1 | 97%
plugins/module_utils/network/ios/rm_templates/l3_interfaces.py | 48 | 0 | 0 | 30 | 2 | 97%
plugins/modules/ios_l3_interfaces.py | 15 | 1 | 0 | 2 | 1 | 88%
tests/unit/modules/network/ios/test_ios_l3_interfaces.py | 78 | 0 | 0 | 0 | 0 | 100%

**Comment by ashwini-mhatre** [comment]
@KB-perByte could you please add integration tests for this?

**Comment by KB-perByte** [comment]
recheck

File: tests/unit/modules/network/ios/test_ios_l3_interfaces.py [hunk 1]
```diff
@@ -257,16 +300,244 @@ def test_ios_l3_interfaces_rendered(self):
...
-            "interface GigabitEthernet0/3",
-            "ip address 192.168.0.3 255.255.255.0",
             "interface GigabitEthernet0/4",
             "ip address 192.168.0.4 255.255.255.0 secondary",
             "interface Serial1/0",
             "ip address 192.168.0.5 255.255.255.0",
         ]
 
         result = self.execute_module(changed=False)
+        print(
```
Comment (ashwini-mhatre) [inline tests/unit/modules/network/ios/test_ios_l3_interfaces.py:314]: remove print statements

**Review by ashwini-mhatre — APPROVED** [review]

File: tests/unit/modules/network/ios/test_ios_l3_interfaces.py (hunk 1 above)
Comment (KB-perByte) [reply tests/unit/modules/network/ios/test_ios_l3_interfaces.py:314]: my bad 👍🏼

**Pull request merged**